clang_format/run_clang_format.sh

```diff
@@ -0,0 +1,33 @@
+#! /bin/bash
+# Usages:
+# run_clang_format format_all <CLANG_FORMAT_BIN> <CLANG_FORMAT_CONFIG>
+# run_clang_format check_file <CLANG_FORMAT_BIN> <INPUT> <OUTPUT>
+set -ue
+
+format_all() {
+    CLANG_FORMAT_CONFIG=$(realpath $1)
+
+    cd $BUILD_WORKSPACE_DIRECTORY
+    if ! test -f .clang-format; then
+        echo ".clang-format file not found. Bazel will copy the default .clang-format file."
+        cp $CLANG_FORMAT_CONFIG .
+    fi
+    git ls-files '*.[ch]' '*.cpp' '*.cc' '*.hpp' | xargs $CLANG_FORMAT_BIN -i
```

Comment:
One last thing before we merge this. This command is a little sluggish and will probably not perform that well in larger projects.

Can we implement is using `diff` like we do for the `clang-format-diff` target in cmake? :https://github.com/swift-nav/cmake/blob/2adba8029f2a1a476b963a32cd494cd03142b6df/ClangFormat.cmake#L238

Reply:
Done